src/bun.js/api/bun/ms.zig

```diff
@@ -0,0 +1,308 @@
+        return std.fmt.allocPrint(allocator, "{d} year{s}", .{ years, if (years != 1) "s" else "" });
+    }
+
+    // Months
+    if (abs_ms >= ms_per_month) {
+        const months = abs_ms / ms_per_month;
+        return std.fmt.allocPrint(allocator, "{d} month{s}", .{ months, if (months != 1) "s" else "" });
+    }
+
+    // Weeks
+    if (abs_ms >= std.time.ms_per_week) {
+        const weeks = abs_ms / std.time.ms_per_week;
+        return std.fmt.allocPrint(allocator, "{d} week{s}", .{ weeks, if (weeks != 1) "s" else "" });
+    }
+
+    // Days
+    if (abs_ms >= std.time.ms_per_day) {
+        const days = abs_ms / std.time.ms_per_day;
+        return std.fmt.allocPrint(allocator, "{d} day{s}", .{ days, if (days != 1) "s" else "" });
+    }
+
+    // Hours
+    if (abs_ms >= std.time.ms_per_hour) {
+        const hours = abs_ms / std.time.ms_per_hour;
+        return std.fmt.allocPrint(allocator, "{d} hour{s}", .{ hours, if (hours != 1) "s" else "" });
+    }
+
+    // Minutes
+    if (abs_ms >= std.time.ms_per_min) {
+        const minutes = abs_ms / std.time.ms_per_min;
+        return std.fmt.allocPrint(allocator, "{d} minute{s}", .{ minutes, if (minutes != 1) "s" else "" });
+    }
+
+    // Seconds
+    if (abs_ms >= std.time.ms_per_s) {
+        const seconds = abs_ms / std.time.ms_per_s;
+        return std.fmt.allocPrint(allocator, "{d} second{s}", .{ seconds, if (seconds != 1) "s" else "" });
+    }
+
+    // Milliseconds
+    return std.fmt.allocPrint(allocator, "{d} ms", .{abs_ms});
+}
+
+/// JavaScript function: Bun.ms(value, options?)
+pub fn jsFunction(
+    globalThis: *JSGlobalObject,
+    callframe: *jsc.CallFrame,
+) JSError!jsc.JSValue {
+    const input, const options = callframe.argumentsAsArray(2);
+
+    // If input is a number, format it to a string
+    if (input.isNumber()) {
+        const ms_value = input.asNumber();
+
+        if (std.math.isNan(ms_value) or std.math.isInf(ms_value)) {
+            return globalThis.throwInvalidArguments("Value must be a finite number", .{});
+        }
+
+        var long = false;
+        if (options.isObject()) {
+            if (try options.get(globalThis, "long")) |long_value| {
+                long = long_value.toBoolean();
+            }
+        }
+
+        const result = try format(bun.default_allocator, ms_value, long);
+
+        var str = String.createExternalGloballyAllocated(.latin1, result);
+        return str.transferToJS(globalThis);
+    }
+
+    // If input is a string, parse it to milliseconds
+    if (input.isString()) {
+        const str = try input.toSlice(globalThis, bun.default_allocator);
+        defer str.deinit();
+
+        const result = parse(str.slice()) orelse std.math.nan(f64);
+        return JSValue.jsNumber(result);
+    }
+
+    return globalThis.throwInvalidArguments("Bun.ms() expects a string or number", .{});
+}
+
+// Bundler macro inlining for Bun.ms
+pub fn astFunction(p: anytype, e_: *const E.Call, loc: logger.Loc) !?Expr {
```

Comment:
To me it feels unnatural that this file is responsible for both JS functions and AST functions. I would separate those out.